app/uk/gov/hmrc/excisemovementcontrolsystemapi/repository/ExciseNumberQueueWorkItemRepository.scala

```diff
@@ -36,19 +40,48 @@ class ExciseNumberQueueWorkItemRepository @Inject()
   collectionName = "excise-number-work-item",
   mongoComponent = mongoComponent,
   itemFormat = ExciseNumberWorkItem.format,
-  workItemFields = WorkItemFields.default,
+  workItemFields = WorkItemFields.default.copy(availableAt = "availableAt", receivedAt = "lastSubmitted"),
   extraIndexes = Seq(
     IndexModel(
-      Indexes.ascending("updatedAt"),
-      IndexOptions().expireAfter(appConfig.getMovementTTL.toSeconds, TimeUnit.SECONDS)
+      Indexes.ascending("lastSubmitted"),
+      IndexOptions().expireAfter(appConfig.workItemTTL.toSeconds, TimeUnit.SECONDS)
+    ),
+    IndexModel(
+      Indexes.ascending("item.exciseNumber"),
+      IndexOptions()
+        .name("item_ern_index")
+        .unique(true)
     )
   )
 ) {
 
   override def now(): Instant = timeService.timestamp()
 
   override val inProgressRetryAfter: Duration = {
-    appConfig.retryAfterMinutes
+    appConfig.workItemInProgressTimeOut.toJava
+  }
+
+  def getWorkItemForErn(ern: String): Future[Seq[WorkItem[ExciseNumberWorkItem]]] = {
+    //Returns a Seq but should only ever be one or zero, as "item.exciseNumber" is a unique index
+
+    collection
```

Comment:
hear you may want to use headOption. This will return a Future[Option[WorkItem[ExciseNumberWorkItem]]]
as we know we only going to get one. This should simplify the function that use it in the service

`collection
      .find(
        in("item.exciseNumber", ern),
      )
      .headOption()`

Reply:
you do not need to use map here, just use headOption as above and this with reutrn a Future[Option[WorkItem]]